enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,454 @@
+while another serves as the combined master/worker node. Between
+installation steps, and after the installer completes, snc.sh makes
+various modifications to the cluster, such as configuring the etcd
+operator to run without HA, tuning CPU, storage, and networking
+details, and deleting unneeded components. Once the cluster is
+properly configured and has stabilized, the createdisk.sh shell script
+takes a snapshot of the master/worker node VM and turns that into the
+VM image that is installed for users by the CRC installer. This
+current SNC image build process works, but has several shortcomings,
+described below:
+
+First, the SNC image build process is very brittle. As changes are
+made in the various OpenShift components, and as new OpenShift
+components are introduced, the SNC build process, or the functionality
+of the resulting CRC cluster, is often broken. The CRC team must then
+debug the issue, find what component change caused it, and make
+changes to the SNC tooling in order to compensate. Even more
+problematic, a component change could be introduced at any time for
+which the current SNC build approach cannot reasonably be made to
+work.
+
+Second, the SNC image build process has very limited control over the
+configuration of the resulting cluster, limiting the ability to
+optimize the cluster’s system resource requirements and functionality
+for CRC’s intended usage. After the OpenShift installer completes, the
+snc.sh script deletes several operators and other components that
+aren’t needed, sets some other operators to unmanaged, and scales
+deployments down to a single replica where possible. Ideally, all
+operators useful in single node developer clusters would run, but
+would be configured to scale their services to a single replica, to
+minimize resource consumption, and to avoid any behavior irrelevant to
+these clusters, while operators that are not useful would not run at
+all. In some cases, operator configuration and behavior might be
+specialized for single node developer clusters, such as a specialized
+networking configuration, a lightweight backend for logging, or
+sending a small subset of telemetry data. These sorts of optimizations
+cannot be implemented externally to the component involved via the
+current SNC build process.
+
+Third, the specific OpenShift component configurations produced via
+the SNC image build process can only be tested in the context of the
+complete CRC product. Since these configurations do not exist within
+the component code repositories, they cannot be tested there, or
+elsewhere, in isolation. This impacts the supportability of CRC
+clusters.
+
+Finally, the single node developer cluster configuration produced by
+the current SNC build process cannot easily be reused in any context
+other than the local VM deployments produced by the CRC
+installer. Similar cluster configurations could be useful in other
+contexts, such as in cloud-based development environments or
```

Comment:
while potentially true, this is outside the scope of this doc.

Reply:
I agree any other specific uses are out of scope. The point I was trying to make in this document was that this enhancement makes the definition of the content of a single-node-developer cluster more orthogonal to the SNC deployment tool used for CRC, and that this could open up opportunities for reuse. Would simply adding a statement that any specific reuse is outside the scope and would need a separate enhancement PR acceptable, or do I need to remove this paragraph along with lines 143-147 below?

Reply:
Added text below.